### PR TITLE
[ESP32] Fix host setup failure on mac and some build docs

### LINKED ISSUE #24734: Unable to compile all-clusters-app on M1 Mac
### Reproduction steps / Feature

With SHA 7237d349526575842a3d111d76aca759abe8d5f8

I am running into "CMake Error" when I execute idf.py

**Steps to reproduce:**

1. cd ~/connectedhomeip
2. git reset --hard 7237d349526575842a3d111d76aca759abe8d5f8
3. git clean -Xdf
4. git submodule update --init
5. export PKG_CONFIG_PATH="/opt/homebrew/opt/openssl@3/lib/pkgconfig"
6. source scripts/bootstrap.sh
7. source scripts/activate.sh
8. cd ~/tools/esp-idf
9. git pull
10. git checkout v4.4.3
11. git submodule update --init
12. ./install.sh
13. . ./export.sh
14. cd ~/connectedhomeip/examples/all-clusters-app/esp32
15. idf.py build

**Step 15 throws the following error:**

```
CMake Error at /Users/priyankahonnakasturi/tools/esp-idf/tools/cmake/build.cmake:452 (message):
Call Stack (most recent call first):
  /Users/priyankahonnakasturi/tools/esp-idf/tools/cmake/project.cmake:384 (idf_build_process)
  CMakeLists.txt:38 (project)


-- Configuring incomplete, errors occurred!
See also "/Users/priyankahonnakasturi/connectedhomeip/examples/all-clusters-app/esp32/build/CMakeFiles/CMakeOutput.log".
cmake failed with exit code 1

```

**Note:** This error has been seen from the following SHA

commit 955f7f5881e575855447917a991d65c995cb4d1b
Author: Shubham Patil <shubham.patil@espressif.com>
Date:  Wed Jan 25 20:41:15 2023 +0530

**Attaching the complete log**
[cmake_error.txt](https://github.com/project-chip/connectedhomeip/files/10538227/cmake_error.txt)

### Platform

darwin

### Platform Version(s)

_No response_

### Type

Platform validated

### (Optional) If manually tested please explain why this is only manually tested

_No response_

### Anything else?

_No response_

### LINKED ISSUE #24816: ModuleNotFound Error: No module named 'lark'
When building the example code lighting app for esp32s3 I get the above mentioned error. Even after running the following command: "source scripts/activate.sh && gn gen out/debug && ninja -C out/debug" 

the output is: 

[48/2051] ACTION //examples/all-cluste...en(//build/toolchain/host:mac_x64_gcc)
FAILED: gen/examples/all-clusters-app/all-clusters-common/app/PluginApplicationCallbacks.h gen/examples/all-clusters-app/all-clusters-common/app/callback-stub.cpp
python3 ../../third_party/pigweed/repo/pw_build/py/pw_build/python_runner.py --gn-root ../../ --current-path ../../examples/all-clusters-app/all-clusters-common --default-toolchain=//build/toolchain/host:mac_x64_gcc --current-toolchain=//build/toolchain/host:mac_x64_gcc --capture-output --python-dep-list-files gen/examples/all-clusters-app/all-clusters-common/all-clusters-common_codegen_codegen_metadata_path_list.txt -- ../../scripts/codegen.py --generator cpp-app --output-dir gen/examples/all-clusters-app/all-clusters-common --expected-outputs gen/examples/all-clusters-app/all-clusters-common/all-clusters-common_codegen.expected.outputs ../../examples/all-clusters-app/all-clusters-common/all-clusters-app.matter
Traceback (most recent call last):
  File "/Users/fjprogramming/connectedhomeip/out/debug/../../scripts/codegen.py", line 28, in <module>
    from matter_idl.matter_idl_parser import CreateParser
  File "/Users/fjprogramming/connectedhomeip/.environment/pigweed-venv/lib/python3.9/site-packages/matter_idl/matter_idl_parser.py", line 7, in <module>
    from lark import Lark
ModuleNotFoundError: No module named 'lark'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/fjprogramming/connectedhomeip/out/debug/../../scripts/codegen.py", line 32, in <module>
    from matter_idl.matter_idl_parser import CreateParser
  File "/Users/fjprogramming/connectedhomeip/.environment/pigweed-venv/lib/python3.9/site-packages/matter_idl/matter_idl_parser.py", line 7, in <module>
    from lark import Lark
ModuleNotFoundError: No module named 'lark'
[57/2051] ACTION //examples/all-cluste...en(//build/toolchain/host:mac_x64_gcc)
ninja: build stopped: subcommand failed.

Does anybody have a solution for this error? I am working on an Intel MacBook.

### PR DESCRIPTION
**Pull request by shubhamdp**
#### Problem
Fixes https://github.com/project-chip/connectedhomeip/issues/24734#issuecomment-1413755055

This may also fix https://github.com/project-chip/connectedhomeip/issues/24816

#### Change Overview
https://github.com/project-chip/connectedhomeip/pull/22071 restricted `requirements.esp32.txt` to only linux platform and hence dependencies required to build the ESP32 examples are not installed on mac.

Restrictions were added because building wheels for `gevent` (dependency of `gdbgui`) failed.

Removed the platform restrictions for all the dependencies other than `gdbgui`. Added some documentation around how to install `gdbgui` on mac.

And some doc fixes. 

#### Testing
- Was able successfully bootstrap and build the esp32 application on mac and Ubuntu.

### CODE REVIEW

**Comment by shubhamdp** [comment]
@andreilitvin I took a suggestion from https://github.com/project-chip/connectedhomeip/issues/23477#issuecomment-1313980840

**Review by andy31415 — APPROVED** [review]

File: docs/guides/esp32/build_app_and_commission.md [hunk 1]
```diff
@@ -53,9 +53,7 @@ functionality can still work fine.
 
 -   Matter
 
-    Before running any other build command, the scripts/activate.sh environment
-    setup script should be sourced at the top level. This script set up a Python
-    environment with libraries used to build and test.
+    Activate the Matter environment.
```
Comment (bzbarsky-apple) [inline docs/guides/esp32/build_app_and_commission.md:56]: I think this should be clearly documented as needing to be run after `export.sh`.

File: docs/guides/esp32/setup_idf_chip.md [hunk 2]
```diff
@@ -57,22 +57,29 @@ step.
...
+
+For MacOS, `gdbgui` python package will not be installed using `bootstrap.sh`
+script as it is restricted only for x64 Linux platforms. It is restricted
+because, building wheels for `gevent` (dependency of `gdbgui`) fails on MacOS.
+
+Workaround is to install `gdbgui` wheels as binary:
 
 ```
-source scripts/bootstrap.sh
+source scripts/activate.sh
```
Comment (bzbarsky-apple) [inline docs/guides/esp32/setup_idf_chip.md:80]: If we just bootstrapped, why do we need to activate?

I assume this step needs to be run after every bootstrap, right?  So perhaps document that after bootstrapping you need to do this part.

File: docs/guides/esp32/setup_idf_chip.md [hunk 3]
```diff
@@ -57,22 +57,29 @@ step.
...
+python3 -m pip install -c scripts/constraints.txt --no-cache --prefer-binary gdbgui==0.13.2.0
+deactivate
```
Comment (bzbarsky-apple) [inline docs/guides/esp32/setup_idf_chip.md:82]: Why is the deactivate needed?

**Review by bzbarsky-apple — APPROVED** [review]

File: docs/guides/esp32/build_app_and_commission.md (hunk 1 above)
Comment (shubhamdp) [reply docs/guides/esp32/build_app_and_commission.md]: This is placed exactly below the section which performs sourcing `esp-idf/export.sh`, still mentioned the same.

File: docs/guides/esp32/setup_idf_chip.md (hunk 2 above)
Comment (shubhamdp) [reply docs/guides/esp32/setup_idf_chip.md]: Oh, I did not know bootstrapping will also activate the environment, removed from here.

Yeah, this part has to be run after every bootstrapping.

File: docs/guides/esp32/build_app_and_commission.md (hunk 1 above)
Comment (bzbarsky-apple) [reply docs/guides/esp32/build_app_and_commission.md]: Right, but nothing says "do these sections in order" (and in fact the ccache thing can be done in any order wrt the other bits and generally will be).  Thank you for the update.

**Review by dhrishi — APPROVED** [review]

**Pull request merged**